server/halomod_app/utils.py

```diff
@@ -38,67 +39,68 @@ def hmf_driver(cls=TracerHaloModel, previous: Union[None, TracerHaloModel] = Non
         return this
 
 
-def create_canvas(objects, q: str, d: dict, plot_format: str = "png"):
+def create_canvas(objects, x_key: str, y_key: str, plot_format: str = "png"):
+    KEYMAP = load_json('../globals/plotData.json')
     # TODO: make log scaling automatic
     fig = Figure(figsize=(10, 6), edgecolor="white", facecolor="white", dpi=100)
     ax = fig.add_subplot(111)
     ax.grid(True)
-    ax.set_xlabel(d["xlab"], fontsize=15)
-    ax.set_ylabel(d["ylab"], fontsize=15)
+    ax.set_xlabel(KEYMAP[x_key]["label"], fontsize=15)
+    ax.set_ylabel(KEYMAP[y_key]["label"], fontsize=15)
 
     lines = ["-", "--", "-.", ":"]
 
-    if q.startswith("comparison"):
+    if y_key.startswith("comparison"):
         compare = True
-        q = q[11:]
+        y_key = y_key[11:]
     else:
         compare = False
 
     # Get the kind of axis we're comparing to.
-    for x, label in XLABELS.items():
-        if KEYMAP[q]["xlab"] == label:
-            break
-    else:
-        raise ValueError(f"The quantity {q} is not found in KEYMAP")
+    # for x, label in XLABELS.items():
+    #     if KEYMAP[q]["xlab"] == label:
+    #         break
+    # else:
+    #     raise ValueError(f"The quantity {y_key} is not found in KEYMAP")
```

Comment:
This can be removed if it is no longer being used.

Reply:
I want to add something to replace that, but it works at the moment and I dont have much time to get that done for this sprint.
A small enhancement for potential error if someone calls the api directly.

Reply:
Okay can a comment be put there for that then? Right now it just looks like commented out code and no one else looking at it could really tell that someone is planning on putting something there.

Reply:
This part has been removed in favor of @vperuzzi's changes